### PR TITLE
InternalSnapshotInfoService refactor

### PR DESCRIPTION
**Pull request by thecoop**
Refactor some internals of `InternalSnapshotInfoService` I spotted during #90006

### CODE REVIEW

**Review by original-brownbear — APPROVED** [review]
LGTM thanks @thecoop !

File: server/src/main/java/org/elasticsearch/snapshots/InternalSnapshotsInfoService.java [hunk 1]
```diff
@@ -65,9 +64,11 @@ public class InternalSnapshotsInfoService implements ClusterStateListener, Snaps
     private final Supplier<RerouteService> rerouteService;
 
     /** contains the snapshot shards for which the size is known **/
+    // volatile for the unlocked access in numberOfKnownSnapshotShardSizes()
+    // map itself is immutable
     private volatile Map<SnapshotShard, Long> knownSnapshotShards;
 
-    private volatile boolean isMaster;
+    private boolean isMaster;
```
[the remark this answers is not in the document]
Comment (thecoop) [reply server/src/main/java/org/elasticsearch/snapshots/InternalSnapshotsInfoService.java:72]: Good idea

**Comment by thecoop** [comment]
@elasticsearchmachine rerun elasticsearch-ci/part-1

**Comment by thecoop** [comment]
@elasticsearchmachine update main

**Review by ChrisHegarty — COMMENTED** [review]
LGTM

**Pull request merged**